homeassistant/components/kulersky/light.py

```diff
@@ -0,0 +1,165 @@
+        await self.hass.async_add_executor_job(self.disconnect)
+
+    def disconnect(self, *args) -> None:
+        """Disconnect the underlying device."""
+        self._light.disconnect()
+
+    @property
+    def name(self):
+        """Return the display name of this light."""
+        return self._light.name
+
+    @property
+    def unique_id(self):
+        """Return the ID of this light."""
+        return self._light.address
+
+    @property
+    def device_info(self):
+        """Device info for this light."""
+        return {
+            "identifiers": {(DOMAIN, self.unique_id)},
+            "name": self.name,
+            "manufacturer": "Brightech",
+        }
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return SUPPORT_KULERSKY
+
+    @property
+    def brightness(self):
+        """Return the brightness of the light."""
+        return self._brightness
+
+    @property
+    def hs_color(self):
+        """Return the hs color."""
+        return self._hs_color
+
+    @property
+    def white_value(self):
+        """Return the white value of this light between 0..255."""
+        return self._white_value
+
+    @property
+    def is_on(self):
+        """Return true if light is on."""
+        return self._brightness > 0 or self._white_value > 0
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._available
+
+    def turn_on(self, **kwargs):
+        """Instruct the light to turn on."""
+        default_hs = (0, 0) if self._hs_color is None else self._hs_color
+        hue_sat = kwargs.get(ATTR_HS_COLOR, default_hs)
+
+        default_brightness = 0 if self._brightness is None else self._brightness
+        brightness = kwargs.get(ATTR_BRIGHTNESS, default_brightness)
+
+        default_white_value = 255 if self._white_value is None else self._white_value
+        white_value = kwargs.get(ATTR_WHITE_VALUE, default_white_value)
+
+        if brightness == 0 and white_value == 0 and not kwargs:
+            # If the light would be off, and no additional parameters were
+            # passed, just turn the light on full brightness.
+            brightness = 255
+            white_value = 255
+
+        rgb = color_util.color_hsv_to_RGB(*hue_sat, brightness / 255 * 100)
+
+        self._light.set_color(*rgb, white_value)
+
+    def turn_off(self, **kwargs):
+        """Instruct the light to turn off."""
+        self._light.set_color(0, 0, 0, 0)
+
+    def update(self):
+        """Fetch new state data for this light."""
+        try:
+            if not self._light.connected:
+                self._light.connect()
+            # pylint: disable=invalid-name
+            r, g, b, w = self._light.get_color()
+        except pykulersky.PykulerskyException as exc:
+            if self._available:
+                _LOGGER.warning(
+                    "Unable to connect to %s", self._light.address, exc_info=exc
```

Comment:
Passing `exc_info` will print a stack trace. That's probably not what we want here, as it's just a connection error. 
```suggestion
                    "Unable to connect to %s: %s", self._light.address, exc
```